network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/waitlist/NetPWaitlistManager.kt

```diff
@@ -58,7 +59,7 @@ class RealNetPWaitlistManager @Inject constructor(
 ) : NetPWaitlistManager {
 
     private val state: MutableStateFlow<NetPWaitlistState> =
-        MutableStateFlow(networkProtectionWaitlist.getState())
+        MutableStateFlow(runBlocking { networkProtectionWaitlist.getState() })
```

Comment:
@karlenDimla , I don't really like this `runBlocking`, do you have a better idea?

Reply:
I agree that `runBlocking` isn't a good addition here. Could we initialise this quickly with a placeholder value (e.g., `loading`) and launch a coroutine to get the actual value which calls `networkProtectionWaitlist.getState()`?

Reply:
@CDRussell I think I found a good solution now.